src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.js

```diff
@@ -46,9 +46,18 @@ export default class ManageEventIndexDialog extends React.Component {
         };
     }
 
-    async updateCurrentRoom(room) {
+    updateCurrentRoom = async (room) => {
         const eventIndex = EventIndexPeg.get();
-        const stats = await eventIndex.getStats();
+        let stats;
+
+        // This call may fail if sporadically, not a huge issue as we will try
+        // later again and probably succeed.
```

Comment:
would be good to move this comment into the `catch` block above the `return` for clarity.